kedro/framework/cli/cli.py

```diff
@@ -96,14 +98,15 @@ def docs():
     webbrowser.open(index_path)
 
 
-def _init_plugins():
-    group = ENTRY_POINT_GROUPS["init"]
-    for entry_point in pkg_resources.iter_entry_points(group=group):
+def _init_plugins() -> None:
+    init_hooks = load_entry_points("init")
+    for init_hook in init_hooks:
         try:
-            init_hook = entry_point.load()
             init_hook()
-        except Exception as exc:
-            raise KedroCliError(f"Initializing {entry_point}") from exc
+        except Exception as exc:  # pylint: disable=broad-except
```

Comment:
Now you've made it clearer exactly what raises the exception here, I actually think we should remove the `try`/`except` altogether and just do a plain
```
for init_hook in init_hooks:
    init_hooks()
```

Reasoning:
* exceptions raised by `entry_point.load()` are now handled by `load_entry_points`
* exceptions raised by `init_hook()` should still be raised, not hidden and logged, and there's not much point wrapping them in `KedroCliError` like we used to. This is analogous to what happens e.g. if a plugin with a hook raises an exception - it doesn't get caught and logged or converted into a kedro error message

All this is kind of minor because realistically no one uses `init_hooks` and it's probably redundant now we have `before_command_run` anyway... But if it's not too annoying to chance I think this would be a small improvement.

Reply:
Good catch, I think it is actually wrong to catch the error, error should be raised!